tests/test_v1.py

```diff
@@ -18,240 +18,275 @@
+    app, _ = mongo_app
     output = app.post(
         "/v1/job", data=job_data, content_type="application/json"
     )
-    job_id = json.loads(output.data.decode()).get("job_id")
+    job_id = output.json.get("job_id")
     assert v1.check_valid_uuid(job_id)
-    # Ensure the queue will expire in about a week (604800s)
-    assert 604000 < testflinger.app.redis.ttl("tf_queue_test") <= 604800
     # Now get the job and confirm it matches
     output = app.get("/v1/job?queue=test")
-    # unittest assertDictContainsSubset is deprecated, but
-    # this works pretty well in its place
+    # Ensure everything we submitted is in the job_data we got back
     expected_data = set(json.loads(job_data))
-    actual_data = set(json.loads(output.data.decode()))
+    actual_data = set(output.json)
     assert expected_data.issubset(actual_data)
 
 
-def test_add_job_good_with_jobid(app):
+def test_add_job_good_with_jobid(mongo_app):
     """Test that adding a job with job ID works"""
     my_id = "77777777-7777-7777-7777-777777777777"
     job_data = json.dumps({"job_id": my_id, "job_queue": "test"})
     # Place a job on the queue
+    app, _ = mongo_app
     output = app.post(
         "/v1/job", data=job_data, content_type="application/json"
     )
-    job_id = json.loads(output.data.decode()).get("job_id")
+    job_id = output.json.get("job_id")
     assert my_id == job_id
 
 
-def test_initial_job_state(app):
+def test_initial_job_state(mongo_app):
     """Ensure initial job state is set to 'waiting'"""
+    app, _ = mongo_app
     job_data = json.dumps({"job_queue": "test"})
     # Place a job on the queue
     output = app.post(
         "/v1/job", data=job_data, content_type="application/json"
     )
-    job_id = json.loads(output.data.decode()).get("job_id")
+    job_id = output.json.get("job_id")
     result_url = "/v1/result/{}".format(job_id)
-    updated_data = json.loads(app.get(result_url).data.decode())
-    assert "waiting" == updated_data.get("job_state")
+    response = app.get(result_url)
+    assert "waiting" == response.json.get("job_state")
 
 
-def test_resubmit_job_state(app):
+def test_resubmit_job_state(mongo_app):
     """Ensure initial job state is set to 'waiting'"""
+    app, _ = mongo_app
     job_data = {"job_queue": "test"}
     # Place a job on the queue
     output = app.post(
         "/v1/job", data=json.dumps(job_data), content_type="application/json"
     )
     # insert the job_id into a job to resubmit
-    job_id = json.loads(output.data.decode()).get("job_id")
+    job_id = output.json.get("job_id")
     job_data["job_id"] = job_id
     output = app.post(
         "/v1/job", data=json.dumps(job_data), content_type="application/json"
     )
     result_url = "/v1/result/{}".format(job_id)
-    updated_data = json.loads(app.get(result_url).data.decode())
-    assert "resubmitted" == updated_data.get("job_state")
+    updated_data = app.get(result_url).json
+    assert "waiting" == updated_data.get("job_state")
```

Comment:
Why do we assert it with `waiting`, not `resubmitted`? I guess it doesn't correspond to the function's title

Reply:
this was a small, intentional change. We used to treat resubmitted as a different state from waiting. In retrospect, there's no real reason to do that. In both cases, the job is still in "waiting" state until something picks it up. So this test doesn't really check for a different job state now, but it does exercise both paths of job submission where there's a job_id already provided, and the one where there's no job_id provided.

Reply:
Got it, thanks